docs/content/doc/advanced/config-cheat-sheet.en-us.md

```diff
@@ -574,6 +574,22 @@ Two special environment variables are passed to the render command:
 - `GITEA_PREFIX_SRC`, which contains the current URL prefix in the `src` path tree. To be used as prefix for links.
 - `GITEA_PREFIX_RAW`, which contains the current URL prefix in the `raw` path tree. To be used as prefix for image paths.
 
+
+Gitea supports customizing the sanitization policy for rendered HTML. The example below will support KaTeX output from pandoc.
+
+```ini
+[markup.sanitizer]
+ELEMENT = span
+ALLOW_ATTR = class
+REGEXP = ^(math\s*|inline\s*|display\s*){0,3}$
```

Comment:
This regexp will match empty strings and the following (removing the quotes):
```
"math"
"math    math    math     "
"math    inline     display   "
"math    
         display   display   "
```
etc.

What is it that you wanted to match?

Maybe what you wanted is ...?:

```suggestion
REGEXP = ^(math|inline|display)$
```

To match either `math`, `inline` or `display`.

Reply:
I at least agree that `{0,3}` shouldn't necessarily be the range, perhaps `{1,2}` is better. In particular, the `pandoc`-rendered Markdown output looks like:

```html
<li><span class="math inline">\{..., -16, -11, -6, -1, 4, 9, 14, ...\}</span></li>
<li><span class="math inline">\{...,-7, -4, -1, 2, 5, 8, ...\}</span></li>
<li><span class="math inline">\{-2, -1, 0, 1, 2, 3, 4, 5, 6\}</span></li>
<li><span class="math inline">\{-1, 0, 1, 2, 3, 4, 5, 6, 7\}</span></li>
<li><span class="math inline">\{-\sqrt{3}, \sqrt{3} \}</span></li>
```

So both classes `math` and (`inline` or `display`) are applied on an element. This matches my understanding from the [pandoc docs](https://pandoc.org/MANUAL.html#math-rendering-in-html). The order seems consistent but isn't guaranteed by that doc. Additionally, classes need to be separated by at least one space, but could be separated by more.

So assuming you want to match any of: `math inline`, `inline math`, `math display`, or `display math` and you aren't worried about duplicate classes (`math math`, `inline inline`, or `display display` -- because while they're not strictly correct there is no security risk in allowing these) and you don't care about whitespace between classes, I **think** the simplest regex you arrive at is:

    ^(math\s*|inline\s*|display\s*){2}$

(and you could argue that there could be another `\s*` at the beginning). However, the docs above seem to imply that unless you pass a rendering agent (`--katex`, `--mathjax` or some such), you might get an element with just `class="math"` -- hence why I'd prefer at _least_ `{1, 2}`.

My understanding from [the `bluemonday`](https://godoc.org/github.com/microcosm-cc/bluemonday#Policy.AllowAttrs) docs is that `.Matching($REGEX)` is a regex applied to the contents of the attribute.

So, if you specify your suggestion of `REGEXP = ^(math|inline|display)$`, my understanding is that it'd sanitize the `class` attribute off of my spans, because they have two classes and not just one. 

Thoughts?

Reply:
> So, if you specify your suggestion of `REGEXP = ^(math|inline|display)$`, my understanding is that it'd sanitize the `class` attribute off of my spans, because they have two classes and not just one.

@cipherboy You're right, I missed the multiple-class case which you conveniently documented in the issue's description. For this then I'd suggest:

```
REGEXP = ^\s*((math(\s+|$)|inline(\s+|$)|display(\s+|$)))+
```

This will exactly match what you want. Otherwise `mathinline` would have been matched.

As a personal reflection, regexps are powerful but they are very difficult to write correctly and have a lot of pitfalls. I know that this was the agreed method, and other parts of Gitea's configuration work this way too, but I would have preferred a simple list of classes (math|inline|display) and build the regexp inside Gitea; less powerful, indeed, but much easier to configure.

Reply:
> Otherwise `mathinline` would have been matched.

@guillep2k Ah no, that's a good catch.

> As a personal reflection, regexps are powerful but they are very difficult to write correctly and have a lot of pitfalls. I know that this was the agreed method, and other parts of Gitea's configuration work this way too, but I would have preferred a simple list of classes `(math|inline|display)` and build the regexp inside Gitea; less powerful, indeed, but much easier to configure.

That's fair. You can also spend as much time as you like making them work. In this case, there's a trade-off between "only matches exactly what I want but is unreadable" and "matches what I want and a few other things but is much more readable". There's probably no harm in matching these other things in this case (we've eliminated the obvious attacks so what remains are largely benign permutations of allowed class names). But, might as well make it correct while we're here.